hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/Gateway.java

```diff
@@ -63,13 +69,35 @@ public static void main(String[] args) throws Exception {
     new Gateway().run(args);
   }
 
+  private void setHttpBaseDir(OzoneConfiguration ozoneConfiguration)
+      throws IOException {
+    if (StringUtils.isEmpty(ozoneConfiguration.get(
+        OzoneConfigKeys.OZONE_HTTP_BASEDIR))) {
+      //Setting ozone.http.basedir if not set so that server setup doesn't
+      // fail.
+      Path tmpMetaDir = Files.createTempDirectory(Paths.get(""),
+          "ozone_s3g_tmp_base_dir");
+      ShutdownHookManager.get().addShutdownHook(() -> {
+        try {
+          FileUtils.deleteDirectory(tmpMetaDir.toFile());
+        } catch (IOException e) {
+          LOG.error("Failed to cleanup temporary S3 Gateway Metadir {}",
+              tmpMetaDir.toFile().getAbsolutePath(), e);
+        }
+      }, 0);
+      ozoneConfiguration.set(OzoneConfigKeys.OZONE_HTTP_BASEDIR,
+          tmpMetaDir.toFile().getAbsolutePath());
```

Comment:
Nit: Please extract `tmpMetaDir.toFile()`, it's executed 3 times.

Reply:
done